gtda/base.py

```diff
@@ -65,14 +72,80 @@ def fit_transform_resample(self, X, y, **fit_params):
         ----------
         X : ndarray of shape (n_samples, ...)
             Input data.
+
         y : ndarray of shape (n_samples, )
             Target data.
 
         Returns
         -------
         Xt : ndarray of shape (n_samples, ...)
             Transformed input.
+
         yr : ndarray of shape (n_samples, ...)
             Resampled target.
+
         """
         return self.fit(X, y, **fit_params).transform_resample(X, y)
+
+
+class PlotterMixin:
+    """Mixin class for all plotters in giotto-tda."""
+
+    def fit_transform_plot(self, X, y=None, sample=0, **plot_params):
+        """Fit to data, then transform and plot a sample in the input collection.
+        Returns the transformed sample input.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, ...)
+            Input data.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        sample : int
+            Sample to be plotted.
+
+        plot_params : dict
+            Optional plotting parameters.
+
+        Returns
+        -------
+        Xt : numpy array of shape (1, ...)
+            Transformed input sample.
+
+        """
+        Xt = self.fit(X, y).transform(X[sample:sample+1], y[sample:sample+1])
+
+        self.plot(Xt, sample=0, **plot_params)
```

Comment:
What about implementing it as fit followed by transform_plot?

Reply:
You transform only what you plot! But you need to fit on everything.

Reply:
Doesn't `transform_plot` also only plot one sample?

Reply:
Implementing as `fit` followed by `transform_plot`.